lockfile/src/lib.rs

```diff
@@ -161,7 +167,7 @@ pub struct Package {
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub enum PackageVersion {
     /// Version from the ecosystem's first-party registry.
-    FirstParty(String),
+    FirstParty(FirstPartyVersion),
```

Comment:
Adding the `PackageType` here is quite a large design decision. We should meet to discuss this. Meeting invite coming soon.

Cc: @cd-work